hoomd/hpmc/ComputeSDF.h

```diff
@@ -364,13 +390,184 @@ template<class Shape> void ComputeSDF<Shape>::countHistogram(uint64_t timestep)
                     }
                 } // end loop over AABB nodes
             }     // end loop over images
+        if (min_bin_ptl_i < m_hist.size())
+            {
+            m_hist[min_bin_ptl_i]++;
+            }
+        } // end loop over all particles
+    }     // end countHistogram()
+
+template<class Shape> void ComputeSDF<Shape>::countHistogramBruteForce(uint64_t timestep)
+    {
+    // update the aabb tree
+    const hoomd::detail::AABBTree& aabb_tree = m_mc->buildAABBTree();
+
+    // update the image list
+    const std::vector<vec3<Scalar>>& image_list = m_mc->updateImageList();
+
+    Scalar extra_width = m_xmax / (1 - m_xmax) * m_mc->getMaxCoreDiameter();
+
+    // access particle data and system box
+    ArrayHandle<Scalar4> h_postype(m_pdata->getPositions(),
+                                   access_location::host,
+                                   access_mode::read);
+    ArrayHandle<Scalar4> h_orientation(m_pdata->getOrientationArray(),
+                                       access_location::host,
+                                       access_mode::read);
+    ArrayHandle<Scalar> h_diameter(m_pdata->getDiameters(),
+                                   access_location::host,
+                                   access_mode::read);
+    ArrayHandle<Scalar> h_charge(m_pdata->getCharges(), access_location::host, access_mode::read);
+    const std::vector<param_type, hoomd::detail::managed_allocator<param_type>>& params
+        = m_mc->getParams();
+
+    // loop through N particles
+    // At the top of this loop, we initialize min_bin_ptl_i to the size of the sdf histogram
+    // For each of particle i's neighbors, we find the scaling that produces the first overlap.
+    // For each neighbor, we do a brute force search from the scaling corresponding to bin 0
+    // up to the minimum bin that we've already found for particle i.
+    // Then we add to m_hist[min_bin_ptl_i] the negative Mayer-function corresponding to the type
+    // of overlap corresponding to particle i's first overlap.
+    for (unsigned int i = 0; i < m_pdata->getN(); i++)
```

Comment:
Why not store result of `getN()` instead of calling on every iteration, like you do with `n_images` below? (Does it change?)

Reply:
`getN` is a const inlined function. The compiler should be able to optimize this away.